src/domain/LottoValidator.js

```diff
@@ -0,0 +1,90 @@
+import { ERROR_MESSAGES } from '../constants/messages.js';
+import {
+  LOTTO_CONDITION,
+  PRICE_MAX_AMOUNT,
+  PRICE_PER_LOTTO,
+} from '../constants/constants.js';
+
+class LottoValidator {
+  validatePurchaseAmount(amount) {
+    if (this.#isEmpty(amount)) throw new Error(ERROR_MESSAGES.empty);
+
+    if (this.#isNotNumeric(amount)) throw new Error(ERROR_MESSAGES.numeric);
+
+    if (this.#isNotPositive(amount)) throw new Error(ERROR_MESSAGES.positive);
+
+    if (this.#isInvalidAmount(amount))
+      throw new Error(ERROR_MESSAGES.invalid_amount);
+
+    if (this.#isExceedRange(amount))
+      throw new Error(ERROR_MESSAGES.exceed_range);
+  }
+
+  validateLottoNumber(numbers) {
+    if (this.#hasInvalidNumber(numbers))
+      throw new Error(ERROR_MESSAGES.invalid_lotto_input);
+
+    if (this.#isInvalidLottoLength(numbers))
+      throw new Error(ERROR_MESSAGES.invalid_lotto_length);
+
+    if (this.#hasOutOfRange(numbers))
+      throw new Error(ERROR_MESSAGES.invalid_lotto_range);
+
+    if (this.#isDuplicate(numbers))
+      throw new Error(ERROR_MESSAGES.invalid_lotto_unique);
+  }
+
+  validateBonusNumber(bonusNumber, isDuplicate) {
+    if (isDuplicate) throw new Error(ERROR_MESSAGES.invalid_lotto_unique);
+
+    if (this.#isNotNumeric(bonusNumber))
+      throw new Error(ERROR_MESSAGES.numeric);
+
+    if (this.#isInvalidLottoRange(bonusNumber))
+      throw new Error(ERROR_MESSAGES.invalid_lotto_range);
+  }
+
+  #isEmpty(input) {
+    return input == null || input.trim() === '';
+  }
+
+  #isNotPositive(input) {
+    return Number(input) <= 0;
+  }
+
+  #isInvalidAmount(input) {
+    return Number(input) % PRICE_PER_LOTTO !== 0;
+  }
+
+  #isExceedRange(input) {
+    return Number(input) > PRICE_MAX_AMOUNT;
+  }
+
+  #isNotNumeric(input) {
+    return Number.isNaN(Number(input));
+  }
+
+  #hasInvalidNumber(numbers) {
+    return numbers.some((number) => this.#isNotNumeric(number));
+  }
+
+  #isInvalidLottoLength(numbers) {
+    return numbers.length !== LOTTO_CONDITION.length;
+  }
+
+  #hasOutOfRange(numbers) {
+    return numbers.some((number) => this.#isInvalidLottoRange(number));
+  }
+
+  #isInvalidLottoRange(number) {
+    return !(
+      number >= LOTTO_CONDITION.startRange && number <= LOTTO_CONDITION.endRange
+    );
```

Comment:
bool 연산자에 개행도 되어 있어서 해석하기 복잡한데 이렇게 표현해보는 건 어떨까요?
```suggestion
    return number < LOTTO_CONDITION.startRange || number > LOTTO_CONDITION.endRange;
```

Reply:
아아... 드모르간 법칙을 생각하지 못했네요..ㅎ 좋은 아이디어 감사합니다!